src/core/server/saved_objects/serialization/types.ts

```diff
@@ -80,3 +81,17 @@ export type SavedObjectUnsanitizedDoc<T = unknown> = SavedObjectDoc<T> & Partial
  * @public
  */
 export type SavedObjectSanitizedDoc<T = unknown> = SavedObjectDoc<T> & Referencable;
+
+/**
+ * Options that can be specified when using the saved objects serializer to parse a raw document.
+ *
+ * @public
+ */
+export interface SavedObjectsRawDocParseOptions {
+  /**
+   * Optional flag to allow for flexible handling of the raw document ID and namespace field. This is needed when a previously
+   * single-namespace object type is converted to a multi-namespace object type, and it is only intended to be used during upgrade
+   * migrations.
+   */
+  flexible?: boolean;
```

Comment:
If this is only meant to be used for single->multi NS switch use cases, I would be more explicit about it in the naming than `flexible`.

Reply:
It is. I couldn't think of a better / more accurate name. Any ideas?

Reply:
I don't love this, but to start the conversation, what about `namespaceTreatment`?

```suggestion
  /**
   * Optional setting to allow for lax handling of the raw document ID and namespace field. This is needed when a previously
   * single-namespace object type is converted to a multi-namespace object type, and it is only intended to be used during upgrade
   * migrations.
   * If not specified, the default treatment is `strict`.
   */
  namespaceTreatment?: 'strict' | 'lax';
```

Reply:
> I don't love this, but to start the conversation, what about `namespaceTreatment`?

I think it's better than `flexible`!

Reply:
+1 for `namespaceTreatment` because I don't really have any better suggestion